src/originalBudgetTrackingApp/export/exportTransactions.ts

```diff
@@ -1,28 +1,42 @@
 import { Config } from '@/originalBudgetTrackingApp/configManager/configManager';
+import { BudgetTrackingEventEmitter, EventNames } from '@/originalBudgetTrackingApp/eventEmitters/EventEmitter';
 import { EnrichedTransaction } from '@/originalBudgetTrackingApp/commonTypes';
 import _ from 'lodash';
 import outputVendors from '@/originalBudgetTrackingApp/export/outputVendors';
 
 export async function createTransactionsInExternalVendors(
   outputVendorsConfig: Config['outputVendors'],
   companyIdToTransactions: Record<string, EnrichedTransaction[]>,
-  startDate: Date
+  startDate: Date, eventEmitter: BudgetTrackingEventEmitter
 ) {
+  await eventEmitter.emit(EventNames.EXPORT_PROCESS_START);
   const executionResult = {};
   const allTransactions = _.flatten(Object.values(companyIdToTransactions));
 
   for (let j = 0; j < outputVendors.length; j++) {
     const outputVendor = outputVendors[j];
     if (outputVendorsConfig[outputVendor.name]?.active) {
+      const baseExporterEventData = { name: outputVendor.name.toString(), allTransactions };
```

Comment:
I think we will prefer the least event emitter code in the flow as we can, right?
I'm just asking, what do you think about hiding the `emit(eventName, payload)` methods by implement methods like `emitExporterStart(name: string, transactions: EnrichedTransactions[])` so you can use one simple line `eventEmitter.emitExporterStart(outputVendor.name.toString(), allTransactions)`?

Of course, this requires a lot of specific code in the `BudgetTrackingEventEmitter`.

Reply:
I think also about setting the `name` of the exporter once so the `eventEmitter` will add it automatically in the related events, like **scopes** in some loggers.
But about this and the idea above, maybe I have good suggestions, but we want to merge this and all, so think carefully, maybe we can forego my suggestions for now.

Reply:
Regarding the first comment. Its a good idea, but it requires one of two:

Either we accept the event name as a param, and then the calls look like this: 
```
await eventEmitter.emitExporterEvent({ eventName: EventNames.EXPORTER_START, outputVendor, allTransactions });
```

Which is not much better than the current situation, or as you suggested we add for every event name a function in the event emitter, which is quite allot of functions. So I think we should leave it as is for now

Reply:
Regarding the second comment, that is also a good idea. I would like to have all this metadata sit in a scope or some other type of state instead of sending it as parameters in every emit call. The complexity here is that I would like the importers to run concurrently, and so maintaining scope that makes sense will probable require something like:

```
eventEmitter.runInScope(scopeParams, () => {
  // Code we want to run
}))
```

Makes the code a but strange, Don't think it is worth it at this point either.

Thoughts or ideas?